src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -135,43 +149,72 @@ private void ConnectToWorkspace(Workspace workspace)
                 ResetLastParsedDocument();
 
                 _workspace = workspace;
-                _workspace.WorkspaceChanged += this.OnWorkspaceChanged;
-                _workspace.DocumentOpened += this.OnDocumentOpened;
-                _workspace.DocumentActiveContextChanged += this.OnDocumentActiveContextChanged;
 
-                var textContainer = _subjectBuffer.AsTextContainer();
+                if (_workspace != null)
+                {
+                    _isClassificationOnlyWorkspace = false;
+                    _workspace.WorkspaceChanged += this.OnWorkspaceChanged;
+                    _workspace.DocumentOpened += this.OnDocumentOpened;
+                    _workspace.DocumentActiveContextChanged += this.OnDocumentActiveContextChanged;
+
+                    var textContainer = _subjectBuffer.AsTextContainer();
 
-                var documentId = _workspace.GetDocumentIdInCurrentContext(textContainer);
-                if (documentId != null)
+                    var documentId = _workspace.GetDocumentIdInCurrentContext(textContainer);
+                    if (documentId != null)
+                    {
+                        var document = workspace.CurrentSolution.GetDocument(documentId);
+                        if (document != null)
+                        {
+                            EnqueueParseSnapshotTask(document);
+                            return;
+                        }
+                    }
+                }
+                else
                 {
-                    var document = workspace.CurrentSolution.GetDocument(documentId);
-                    if (document != null)
+                    // This buffer isn't being tracked by any workspace. In Visual Studio, this
+                    // can happen if it is not in the Running Document Table. We'll try to do
+                    // basic syntax classification anyway.
+
+                    _isClassificationOnlyWorkspace = true;
+                    var languageName = _contentTypesToLanguageNames.FirstOrDefault(x => _subjectBuffer.ContentType.MatchesAny(x.Metadata.DefaultContentType))?.Metadata.Language;
+                    if (languageName != null)
                     {
+                        _workspace = new AdhocWorkspace();
```

Comment:
We're not passing MEF host services here? That'll result in another MEF container.